### DISCUSSION TITLE
An unexcepted differences in energy calculations with/without LAMMPS

### DISCUSSION THREAD

**Discussion by DemingXia** [deepmodeling/deepmd-kit#1620] [General]
Dear all,

I trained a DeepMD model (named model_07.gb) with acceptable RMSE in energy (ca. 0.4 meV) and force (ca. 90 meV/Angstrom) using the attached input.txt. 
[input.txt](https://github.com/deepmodeling/deepmd-kit/files/8412294/input.txt)

Then, I made a test in two different cases for a single structure:
(1) Calculated energy using the model directly, which was achieved by the command "dp test -m model_07.gb -s "path/to/system/for/input/structure" -d XXX". 
(2) Calculated energy using "LAMMPS + the model" for the SAME structure, which was achieved by running the attached input file for LAMMPS.
[lammps.txt](https://github.com/deepmodeling/deepmd-kit/files/8412298/lammps.txt)

I think that the energies obtained in the above two cases should be the same or at least with a tiny error. However, I found a huge difference (ca. 24 eV) between the calculated energies in the above two cases (-139128.1320 eV versus -139104.5760 eV). I also tested many different structures, but all of them showed huge differences.
Could someone please help me with the potential reason for the errors (e.g., wrong inputs for lammps)?
Much thanks for the potential clues and reply!

Deming Xia

**Comment by njzjz** [comment]
Could you provide an example to reproduce it?

**Reply by DemingXia** [reply]
Thanks for the reply! As mentioned by Han Wang, the printed output of LAMMPS is the total energy (etotal) that can not be directly compared with potential energy. I have checked the potential energy. The problem is solved! Thanks again!

**Comment by wanghan-iapcm** [comment]
From the lammps script I see you generated the velocity at T=300K and printed total energy. A difference between the potential energy and the total energy is expected.
Please print the potential energy from lammps and then compare it with the DP's python interface.

**Reply by DemingXia** [reply]
Thanks for the reply! I have checked the potential energy of LAMMPS and the DP's python interface. They are the same. Thanks again~
